url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FisImporter.java

```diff
@@ -0,0 +1,626 @@
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+package com.fuzzylite.imex;
+
+import com.fuzzylite.Engine;
+import com.fuzzylite.Op;
+import com.fuzzylite.defuzzifier.Bisector;
+import com.fuzzylite.defuzzifier.Centroid;
+import com.fuzzylite.defuzzifier.LargestOfMaximum;
+import com.fuzzylite.defuzzifier.MeanOfMaximum;
+import com.fuzzylite.defuzzifier.SmallestOfMaximum;
+import com.fuzzylite.defuzzifier.WeightedAverage;
+import com.fuzzylite.defuzzifier.WeightedSum;
+import com.fuzzylite.factory.FactoryManager;
+import com.fuzzylite.hedge.Any;
+import com.fuzzylite.hedge.Extremely;
+import com.fuzzylite.hedge.Not;
+import com.fuzzylite.hedge.Seldom;
+import com.fuzzylite.hedge.Somewhat;
+import com.fuzzylite.hedge.Very;
+import com.fuzzylite.norm.s.AlgebraicSum;
+import com.fuzzylite.norm.s.BoundedSum;
+import com.fuzzylite.norm.s.DrasticSum;
+import com.fuzzylite.norm.s.EinsteinSum;
+import com.fuzzylite.norm.s.HamacherSum;
+import com.fuzzylite.norm.s.Maximum;
+import com.fuzzylite.norm.s.NormalizedSum;
+import com.fuzzylite.norm.t.AlgebraicProduct;
+import com.fuzzylite.norm.t.BoundedDifference;
+import com.fuzzylite.norm.t.DrasticProduct;
+import com.fuzzylite.norm.t.EinsteinProduct;
+import com.fuzzylite.norm.t.HamacherProduct;
+import com.fuzzylite.norm.t.Minimum;
+import com.fuzzylite.rule.Rule;
+import com.fuzzylite.rule.RuleBlock;
+import com.fuzzylite.term.Bell;
+import com.fuzzylite.term.Constant;
+import com.fuzzylite.term.Discrete;
+import com.fuzzylite.term.Function;
+import com.fuzzylite.term.Gaussian;
+import com.fuzzylite.term.GaussianProduct;
+import com.fuzzylite.term.Linear;
+import com.fuzzylite.term.PiShape;
+import com.fuzzylite.term.Ramp;
+import com.fuzzylite.term.Rectangle;
+import com.fuzzylite.term.SShape;
+import com.fuzzylite.term.Sigmoid;
+import com.fuzzylite.term.SigmoidDifference;
+import com.fuzzylite.term.SigmoidProduct;
+import com.fuzzylite.term.Term;
+import com.fuzzylite.term.Trapezoid;
+import com.fuzzylite.term.Triangle;
+import com.fuzzylite.term.ZShape;
+import com.fuzzylite.variable.InputVariable;
+import com.fuzzylite.variable.OutputVariable;
+import com.fuzzylite.variable.Variable;
+import java.io.BufferedReader;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ *
+ * @author jcrada
+ */
+public class FisImporter extends Importer {
+
+    protected static int and = 0, or = 1, imp = 2, agg = 3, defuzz = 4;
+
+    @Override
+    public Engine fromString(String fis) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.